sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -1300,14 +1301,16 @@ trait ComplexTypeMergingExpression extends Expression {
   lazy val inputTypesForMerging: Seq[DataType] = children.map(_.dataType)
 
   def dataTypeCheck: Unit = {
-    require(
-      inputTypesForMerging.nonEmpty,
-      "The collection of input data types must not be empty.")
-    require(
-      TypeCoercion.haveSameType(inputTypesForMerging),
-      "All input types must be the same except nullable, containsNull, valueContainsNull flags. " +
-        s"The expression is: $this. " +
-        s"The input types found are\n\t${inputTypesForMerging.mkString("\n\t")}.")
+    SparkException.require(
+      requirement = inputTypesForMerging.nonEmpty,
+      errorClass = "COMPLEX_EXPRESSION_UNSUPPORTED_INPUT.NO_INPUTS",
```

Comment:
@hannahkamundson Just in case, can users get the error using public API? If so, please, add a test. If not, we should think of converting them to internal errors.

Reply:
I'm pretty sure they are public facing because they are called form things like `Least`. I'll add tests

Reply:
> I'll add tests

Please, add end-to-end tests for the errors to `QueryCompilationErrorsSuite` using public API or SQL, and remove the tests from `ExpressionSuite`.

FYI, we introduce error classes for user facing errors. In the case if an error is not visible to users, we should consider to convert it to an internal error.